library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/ExtensionSupport.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay
+
+import android.view.View
+import com.datadog.android.sessionreplay.internal.recorder.mapper.WireframeMapper
+
+/**
+ * In case you need to provide different configuration for a specific Android UI framework that
+ * is not supported by our SR instrumentation layer (e.g Material elements)
+ * you can implement this class.
+ * @see [SessionReplayConfiguration.Builder.addExtensionSupport]
+ */
+interface ExtensionSupport {
+    /**
+     * Use this method if you want to apply a custom [WireframeMapper] for a specific [View] type
+     * and for a specific [SessionReplayPrivacy].
+     * @return the [View] type to custom [WireframeMapper] map grouped by [SessionReplayPrivacy].
+     */
+    fun getCustomViewMappers(): Map<SessionReplayPrivacy, Map<Class<*>, WireframeMapper<View, *>>>
```

Comment:
side note: I'm curious if `Wireframe` mapper would have `SessionReplayPrivacy` as `map` method argument. Would it make architecture better or worse? wdyt?

Reply:
oh...that will make it worse as i will introduce a new state in the `WireframeMapper`. At least this way I am benefiting from the strategy pattern.